apps/sms/views/shared/js/navigation.js

```diff
@@ -1,361 +1,726 @@
-/* global Promise,
-      EventDispatcher
-*/
-
-/* exported Navigation */
+/*global
+  Utils
+ */
+(function(exports) {
 
 'use strict';
 
-(function(window) {
+/**
+ * The Defer type is useful when creating promises.
+ * @typedef {Object} Defer
+ * @property {function(*)} resolve The Promise's resolve function.
+ * @property {function(*)} reject The Promise's reject function.
+ * @property {Promise} promise The actual promise.
+ */
+
+/**
+ * @function
+ * The debug function translates to console.log in debug mode.
+ *
+ * @param {...*} args The values to be displayed to the console.
+ */
+const debug = 1 ? (...args) => console.log(...args) : () => {};
```

Comment:
At first I was thinking that it should be like snippet below to easily see source of the debug message, but then it won't be possible to use ```debug``` with placeholders like ```debug('Init(%s)', param)```. So it's better like you did :)
```
var debug = 1 ? (...args) => console.log('[Navigation]', ...args) : () => {};
```